code/obj/effects/particles.dm

```diff
@@ -368,12 +368,26 @@
 	gravity = list(0, 0, 0)
 	friction = generator("num", 0.8, 0.6, UNIFORM_RAND)
 
-/obj/effects/gunshot_impact
+/obj/effects/impact
+	var/static/effect_amount = 0
+
+	New()
+		effect_amount ++
+		if (effect_amount > 450) //About 150 gunshots in the span of 5 seconds
+			qdel(src)
+		. = ..()
```

Comment:
Would rather this was done in the logic that spawns these, spawning then instantly deleting something still incurs a performance cost.

Reply:
Moved it out of the particles and into the bullets, so we dont spawn and instantly delete the particles.